src/spriteloader/grf.cpp

```diff
@@ -191,23 +192,22 @@ bool DecodeSingleSprite(SpriteLoader::Sprite *sprite, SpriteFile &file, size_t f
 		dest = dest_orig.get();
 
 		for (int i = 0; i < sprite->width * sprite->height; i++) {
-			byte *pixel = &dest[i * bpp];
+			byte &pixel = dest[i * bpp];
 
 			if (colour_fmt & SCC_RGB) {
-				sprite->data[i].r = *pixel++;
-				sprite->data[i].g = *pixel++;
-				sprite->data[i].b = *pixel++;
+				sprite->data[i].r = pixel++;
+				sprite->data[i].g = pixel++;
+				sprite->data[i].b = pixel++;
```

Comment:
These are not equivalent.

`*pixel++` reads the value pointed to, then increments the pointer to point at the next value.
Your change makes it increment the pointed-to value and does not move the pointer.

Reply:
fixed